docs/source/starthere/fundamentals.rst

```diff
@@ -0,0 +1,240 @@
+
+This will use the default config file specified inside the ``@hydra_runner`` decorator.
+
+We can specify a different config file to use by calling the script like this:
+
+.. code-block:: diff
+
+	 python run_example_training.py \
+	+    --config_path="different_config_file_dir_path" \
+	+    --config_name="different_config_file_name"
+
+We can also override, delete or add elements to the config when we call the script like this:
+
+
+.. code-block:: diff
+
+	 python run_example_training.py \
+	     --config_path="different_config_file_dir_path" \
+	     --config_name="different_config_file_name" \
+	+    model.optim.lr=0.001 \                                     # overwriting
+	+    model.train_ds.manifest_filepath="your_train_data.json" \  # overwriting
+	+    ~trainer.max_epochs \                                      # deleting
+	+    +trainer.max_steps=1000                                    # adding
+
+Running NeMo scripts
+--------------------
+
+NeMo scripts typically take on the form shown above, where the Python script relies on a config object which has some specified default values that you can choose to override.
+
+The NeMo `examples <https://github.com/NVIDIA/NeMo/tree/main/examples/>`__ directory contains many scripts for training and inference of various existing NeMo models. Note that this includes default configs whose default values for model, optimizer and trainer parameters were tuned over the course of many GPU-hours of the NeMo team's experiments. We thus recommend using these as a starting point for your own experiments.
+
+.. note::
+	**NeMo inference scripts**
+
+	The examples scripts directory also contains many inference scripts, e.g. `transcribe_speech.py <https://github.com/NVIDIA/NeMo/blob/main/examples/asr/transcribe_speech.py>`_. These normally have a different structure to the training scripts, as they have a lot of additional utilities for reading and saving files. The inference scripts also use configs, but these naturally do not require the ``trainer``, ``model``, ``exp_manager`` sections. Additionally, due to having fewer elements, the default configs for inference scripts are normally specified as dataclasses rather than separate files. Elements also can be overwritten/added/deleted via the command line.
+
+
+Specifying training data
+------------------------
+
+NeMo will handle creation of data loaders for you, as long as you put your data into the expected input format. You may also need to train a tokenizer before starting training. Learn more about data formats for :doc:`LLM <../nlp/nemo_megatron/gpt/gpt_training>`, :doc:`Multimodal <../multimodal/mllm/datasets>`, :ref:`Speech AI <section-with-manifest-format-explanation>`, and :doc:`Vision models <../vision/datasets>`.
+
+
+Model checkpoints
+-----------------
+
+Throughout training, model checkpoints will be saved inside ``.nemo`` files. These are archive files containing all the necessary components to restore a usable model, e.g.:
```

Comment:
link to https://docs.nvidia.com/nemo-framework/user-guide/latest/nemotoolkit/checkpoints/intro.html

Reply:
added a link `` :doc:`checkpoints <../checkpoints/intro>` ``. It doesn't currently work because that doc does not exist in this branch, but will exist in future.